src/Symfony/Component/Cache/Traits/FilesystemCommonTrait.php

```diff
@@ -107,14 +108,20 @@ private function write(string $file, string $data, int $expiresAt = null)
             }
             fwrite($h, $data);
             fclose($h);
+            $unlink = true;
 
             if (null !== $expiresAt) {
                 touch($this->tmp, $expiresAt ?: time() + 31556952); // 1 year in seconds
             }
 
-            return rename($this->tmp, $file);
+            return !$unlink = !rename($this->tmp, $file);
         } finally {
             restore_error_handler();
+
+            if ($unlink) {
+                @unlink($this->tmp);
+                $this->tmp = null;
```

Comment:
If we reset it to `null` here, is there any benefit at having it as an instance property rather than a local variable ?

Reply:
the instance property was meant to avoid re-reading random bytes, but this was a benefit only if it was kept for next usage.

Reply:
we reset to null only if rename fails

Reply:
which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit

Reply:
> which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit

We will probably not hit it, as we just unlink the temporary file.

Reply:
> > which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit
> 
> We will probably not hit it, as we just unlink the temporary file.

I mean, if we reach tis point, it means that we were able to create a temporary file, write something in it, but moving it fails. It is highly probable that we will be able to link it, as moving failure is probably due to a restriction on the target path, not on the source path.